src/main/java/qna/domain/Question.java

```diff
@@ -1,115 +1,112 @@
     private boolean deleted = false;
 
-    @OneToMany(mappedBy = "question")
-    private List<Answer> answers = new ArrayList<>();
-
-    public Question(String title, String contents) {
-        this(null ,title, contents);
-    }
-
-    public Question(Long id, String title, String contents) {
-        this.id = id;
-        this.title = title;
-        this.contents = contents;
-    }
-
-    public Question(User writer, String title, String contents, boolean deleted) {
+    public Question(User writer, String title, String contents) {
+        validation(writer);
         this.writer = writer;
         this.title = title;
         this.contents = contents;
-        this.deleted = deleted;
     }
 
-    protected Question() {}
-
-    public Question writeBy(User writer) {
-        this.writer = writer;
-        return this;
-    }
-
-    public boolean isOwner(User writer) {
-        return this.writer.equals(writer);
+    protected Question() {
     }
 
-    public void addAnswer(Answer answer) {
-        answer.toQuestion(this);
+    private void validation(User writer) {
+        if (isEmpty(writer)) {
+            throw new UnAuthorizedException("질문 작성자가 없습니다.");
+        }
     }
 
     public Long getId() {
         return id;
     }
 
-    public void setId(Long id) {
-        this.id = id;
-    }
-
-    public String getTitle() {
-        return title;
+    public User getWriter() {
+        return writer;
     }
 
-    public void setTitle(String title) {
-        this.title = title;
+    public boolean isOwner(User writer) {
+        return this.writer.equals(writer);
     }
 
-    public String getContents() {
-        return contents;
+    public void addAnswer(Answer answer) {
+        answers.add(answer);
     }
 
-    public void setContents(String contents) {
-        this.contents = contents;
+    public boolean isDeleted() {
+        return deleted;
     }
 
-    public User getWriter() {
-        return writer;
+    public DeleteHistory delete(User loginUser) {
+        if (!isOwner(loginUser)) {
+            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
+        }
+        if (isDeleted()) {
+            throw new CannotDeleteException("이미 삭제된 질문입니다.");
+        }
+        changeDeletedStatus(true);
+        return new DeleteHistory(ContentType.QUESTION, id, writer);
     }
 
-    public void setWriter(User writer) {
-        this.writer = writer;
+    public List<DeleteHistory> deleteQuestionByAnswers(User loginUser, List<Answer> deleteAnswers) {
+        List<DeleteHistory> result = new ArrayList<>();
+        for (Answer answer : deleteAnswers) {
+            if (!answer.isOwner(loginUser)) {
+                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
+            }
+            result.add(answer.delete(loginUser));
+        }
+        return result;
     }
 
-    public boolean isDeleted() {
-        return deleted;
+    private void changeDeletedStatus(boolean deleted) {
```

Comment:
현재 true 값으로 설정하는 용도로만 사용하고 있는데, 매개변수를 받지 않고 메서드 시그니처를 `ex) changeDeletedTrue()` 처럼 사용하는건 어떻게 생각하시나요?
```suggestion
    private void changeDeletedTrue() {
        this.deleted = true;
    }
```

Reply:
제가 상태값을 매개변수로 받으려고 했던 이유는 삭제 처리된 건을 다시 복구 처리를 해야하는 
로직에 대해 고민하고 결론을 내었는데요😊
그런데 막상 다시 생각해보니 유지보수 관점에선 `changeDeletedTrue `와 `changeDeletedFalse` 처럼
직관적으로 작성하는것이 좋겠다고 생각되네요😂
피드백 감사합니다!!👍👍